test/isolated/config.test.js

```diff
@@ -33,6 +39,12 @@ describe('config', () => {
     expect(config.webpack).toBe(null)
   })
 
+  it('Should assign defaults deeply to customConfig', () => {
```

Comment:
We should really deprecate customConfig, it makes no sense to have it vs next.config.js.

Reply:
But that's a different story than this PR.

Reply:
It would be important for use to keep this. We have a system that constructs and passes in the the next config for our apps and do not use a next.config.js file directly.

Reply:
+1 to the comment from @kinetifex – removing `customConfig` would be a serious issue for us as consumers of `next`.

Reply:
Why though, it causes inconsistency, for example the config is not shared with `next build`

Reply:
The core functionality offered by Next is fantastic for meeting our needs, but we do not use the built-in `next` CLI. We have a separate CLI with a more feature rich plugin system – similar to plugins & presets offered by `vue` CLI. 

The pattern I have observed over time with many large projects that ship CLIs is that the core functionality is eventually separated from the CLI. Examples of this are `babel`, `webpack` to name a few. 

With that in mind I would advise avoiding making decisions that force users to **only** use the `next` CLI as I would imagine it too will eventually be separate from the core `next` functionality.

Reply:
We have plugin system with a cli to spin up and run our web apps, we don't invoke the `next` scripts directly. With our system, we have a plugin that sets up Next.js as a component of the app, assembles the customConfig from our other plugins, and passes that to `next` for startup as in [some custom server examples](https://github.com/zeit/next.js/blob/6b7864e57e481efb26e85e2ecb44e98d336be5ec/examples/with-firebase-hosting-and-docker/src/functions/index.js#L10). We also pass the config to [`build`](https://github.com/zeit/next.js/blob/9c4eefcdbf4058b65c8b88fe98387166acad127b/packages/next/build/index.ts#L43) as well, though we have to require it from `dist`. It would be ideal if this was exported by the package.

Reply:
I see, thanks for explaining! I hadn't seen such a use case before 👍